pandas/io/excel/_openpyxl.py

```diff
@@ -412,7 +419,25 @@ def write_cells(
         _style_cache: Dict[str, Dict[str, Serialisable]] = {}
 
         if sheet_name in self.sheets:
-            wks = self.sheets[sheet_name]
+            if "r+" in self.mode:
+                if self.if_sheet_exists == "new":
+                    wks = self.book.create_sheet()
+                    # openpyxl will create a name for the new sheet by appending digits
+                    wks.title = sheet_name
+                    self.sheets[wks.title] = wks
+                elif self.if_sheet_exists == "replace":
+                    wks = self.sheets[sheet_name]
+                    wks.delete_cols(1, wks.max_column)
```

Comment:
What happens to formatting with this? E.g. I don't know if there is sheet-level formatting as well as cell-level formatting in xlsx, or if anything else might be left behind. Is it safer/faster to use `del self.book[sheet_name]` instead? Still need to also `del self.sheets[sheet_name]` and then recreate with this.

Reply:
Good question. [This link](https://support.microsoft.com/en-us/office/change-the-font-style-and-size-for-a-worksheet-b3f1792b-9980-4b92-9aa5-5dd6e940b195) suggests that there is no such thing as whole worksheet formatting. I've just done a few tests and it looks like things like cell formatting, conditional formatting and column widths are all reset by the `delete_cols` call.

Deleting and recreating the sheet would almost certainly be faster though and there may be something I've missed, so it's probably the way to go.

To get the recreated sheet in the right position I think the code would look like this:
```
old_wks = self.sheets[sheet_name]
target_index = self.book.index(old_wks)
del self.book[sheet_name]
wks = self.book.create_sheet(sheet_name, target_index)
self.sheets[sheet_name] = wks
```

I have some fairly involved examples at work so will do some tests tomorrow and make sure this doesn't cause any issues.